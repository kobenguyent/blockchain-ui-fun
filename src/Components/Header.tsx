import {Button, Container, Navbar} from "react-bootstrap";
import appLogo from '../../src/assets/kobecoin.png'

export const Header = ({ setTxCreation, setShowPendingTxList, pendingTxList }:any) => {
    return (
        <Navbar bg="light" expand="lg">
            <Container>
                <Navbar.Brand onClick={() => {
                    setShowPendingTxList(false)
                    setTxCreation(false)
                    window.location.reload()
                }}><img
                    src={appLogo}
                    width="130"
                    height="50"
                    className="d-inline-block align-top"
                /></Navbar.Brand>
                <Container>
                    <Button variant="outline-secondary" onClick={() => {
                        setShowPendingTxList(true)
                        setTxCreation(false)
                    }}>Pending Transactions ({pendingTxList.length})</Button>
                    <Button variant="outline-secondary" onClick={() => setTxCreation(true)}>Create Transaction</Button>
                </Container>
            </Container>
        </Navbar>
    )
}
