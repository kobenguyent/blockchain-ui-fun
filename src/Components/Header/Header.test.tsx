import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import {Header} from "./Header.tsx";

describe('Header component', async () => {
    it('Should render the header component correctly', async () => {
        // Setup
        render(<Header/>);

        const navBrand = await screen.getByTestId('logo')
        const showPendingTxBtn = await screen.getByTestId('showPendingTransactions')
        const createTxBtn = await screen.getByTestId('createTransaction')

        expect(navBrand).toMatchSnapshot()
        expect(showPendingTxBtn).toMatchSnapshot()
        expect(createTxBtn).toMatchSnapshot()
    });
});
