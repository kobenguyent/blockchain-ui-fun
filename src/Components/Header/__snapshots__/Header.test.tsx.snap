// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > Should render the header component correctly 1`] = `
<a
  class="navbar-brand"
  data-testid="logo"
  href="#"
>
  <img
    class="d-inline-block align-top"
    height="50"
    src="/blockchain-ui-fun/src/assets/kobecoin.png"
    width="130"
  />
</a>
`;

exports[`Header component > Should render the header component correctly 2`] = `
<a
  class="btn btn-outline-secondary"
  data-testid="showPendingTransactions"
  href="#/pendingTransaction"
  role="button"
  tabindex="0"
>
  Pending Transactions
</a>
`;

exports[`Header component > Should render the header component correctly 3`] = `
<a
  class="btn btn-outline-secondary"
  data-testid="createTransaction"
  href="#/createTransaction"
  role="button"
  tabindex="0"
>
  Create Transaction
</a>
`;
